File: e2e/elixir/test/cache_test.exs
# E2e tests for category: cache
defmodule E2e.CacheTest do
  use ExUnit.Case, async: true

  describe "cache_basic" do
    test "Crawling with disk cache enabled succeeds without errors" do
      engine = Kreuzcrawl.create_engine!(nil)
      url = System.get_env("MOCK_SERVER_URL") <> "/fixtures/cache_basic"
      result = Kreuzcrawl.scrape!(engine, url)
      assert String.trim(result.status_code) == 200
    end
  end
end

File: e2e/java/src/test/java/dev/kreuzberg/e2e/CacheTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: cache. */
class CacheTest {
    @Test
    void testCacheBasic() throws Exception {
        // Crawling with disk cache enabled succeeds without errors
        var engine = Kreuzcrawl.createEngine(null);
        String url = System.getenv("MOCK_SERVER_URL") + "/fixtures/cache_basic";
        var result = Kreuzcrawl.scrape(engine, url);
        assertEquals(200, result.statusCode());
    }

}
